### PR TITLE
feat:add sdk module

### PR DESCRIPTION
**Pull request by goldenxinxing**
## Modules
- [ ] UI
- [x] Controller
- [x] Agent
- [ ] Client
- [ ] Python-SDK
- [ ] Others

## Checklist
- [ ] run code format and lint check
- [ ] add unit test
- [x] add necessary doc

### CODE REVIEW

File: server/controller-sdk/pom.xml [hunk 1]
```diff
@@ -3,10 +3,12 @@
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>server</artifactId>
+        <groupId>ai.starwhale</groupId>
+        <version>1.0-SNAPSHOT</version>
```
Comment (tianweidut) [inline server/controller-sdk/pom.xml:9]: starwhale version? use 0.1.0 ?

File: server/controller/pom.xml [hunk 2]
```diff
@@ -3,27 +3,14 @@
...
-    <version>1.0-SNAPSHOT</version>
-
     <parent>
-        <groupId>org.springframework.boot</groupId>
-        <artifactId>spring-boot-starter-parent</artifactId>
-        <version>2.6.4</version>
-        <relativePath/> <!-- lookup parent from repository -->
+        <artifactId>server</artifactId>
+        <groupId>ai.starwhale</groupId>
+        <version>1.0-SNAPSHOT</version>
```
Comment (tianweidut) [inline server/controller/pom.xml:9]: `1.0-SNAPSHOT` -> `0.1.0-SNAPSHOT` ?

File: server/agent-sdk/pom.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <artifactId>server</artifactId>
+        <groupId>ai.starwhale</groupId>
+        <version>1.0-SNAPSHOT</version>
```
Comment (tianweidut) [inline server/agent-sdk/pom.xml:9]: use `0.1.0-SNAPSHOT` ?

File: server/pom.xml [hunk 4]
```diff
@@ -0,0 +1,302 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>ai.starwhale</groupId>
+    <artifactId>server</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (tianweidut) [inline server/pom.xml:9]: use `0.1.0-SNAPSHOT` ?

**Comment by tianweidut** [comment]
@goldenxinxing  maybe you can set PR labels、Project etc.
image

**Review by tianweidut — CHANGES_REQUESTED** [review]

File: server/controller-sdk/pom.xml (hunk 1 above)
Comment (goldenxinxing) [reply server/controller-sdk/pom.xml]: done

**Comment by goldenxinxing** [comment]
> @goldenxinxing maybe you can set PR labels、Project etc. image

ok

**Comment by anda-ren** [comment]
LGTM

**Review by anda-ren — APPROVED** [review]
LGTM

File: server/agent-sdk/pom.xml (hunk 3 above)
Comment (goldenxinxing) [reply server/agent-sdk/pom.xml]: done

File: server/pom.xml (hunk 4 above)
Comment (goldenxinxing) [reply server/pom.xml]: done

File: server/controller/pom.xml (hunk 2 above)
Comment (goldenxinxing) [reply server/controller/pom.xml]: done

**Review by tianweidut — APPROVED** [review]

**Comment by tianweidut** [comment]
LGTM

**Pull request merged**